# Health Data Analysis & Statistical Testing

## Objective

The objective of this project is to perform statistical analysis and hypothesis
testing on a health dataset using Python.

The analysis includes:
- Inferential Statistics
- Hypothesis Testing
- Confidence Interval
- Critical Value
- p-value
- t-test
- Chi-square Test
- ANOVA
- Covariance
- Correlation

## Import Libraries

The following Python libraries are used for data manipulation and statistical analysis.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from scipy import stats

## Load Dataset
The health dataset is loaded using Pandas.

In [2]:
# Load dataset
df = pd.read_csv(r"C:\Users\DDR\Downloads\health_dataset_records.csv")

# Display first 5 records
df.head()

,record_id,age_group,age,weight,gender,region,smoking_status,exercise_frequency,bmi,blood_pressure,diabetes,hypertension,cholesterol_level,glucose_level,visit_date
0,33046596-905d-4076-ae2e-e997b0bb0f62,46-60,56,81,Female,South,Smoker,Weekly,29.4,150.4,False,False,295.1,114.4,09-05-2025
1,b5771b91-b626-45e9-9cb8-ff571da64d1a,60+,69,59,Male,South,Smoker,Weekly,22.2,148.3,False,True,207.6,104.5,08-02-2024
2,590ec651-eed8-467e-9e24-1b3c74610956,46-60,46,48,Female,South,Former Smoker,Weekly,19.3,144.0,False,False,227.3,97.4,10-06-2024
3,4d1af195-1f03-45ac-b873-7896a31ef1e9,26-35,32,66,Female,West,Smoker,Weekly,22.0,130.2,False,False,237.3,103.7,30-04-2024
4,15209345-f431-4899-b8fc-84501b40442b,46-60,60,62,Male,East,Non-Smoker,Weekly,26.2,148.3,True,False,247.1,127.5,02-02-2025


 ## Explore the Dataset
 Before performing statistical tests, we check the number of rows,
columns, column names, data types, and missing values.

In [3]:
# Number of rows and columns
print("Dataset Shape:", df.shape)

# Column names
print("\nColumn Names:")
print(df.columns.tolist())

# Dataset information
print("\nDataset Information:")
df.info()

Dataset Shape: (1000, 15)

Column Names:
['record_id', 'age_group', 'age', 'weight', 'gender', 'region', 'smoking_status', 'exercise_frequency', 'bmi', 'blood_pressure', 'diabetes', 'hypertension', 'cholesterol_level', 'glucose_level', 'visit_date']

Dataset Information:
<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   record_id           1000 non-null   str    
 1   age_group           1000 non-null   str    
 2   age                 1000 non-null   int64  
 3   weight              1000 non-null   int64  
 4   gender              1000 non-null   str    
 5   region              1000 non-null   str    
 6   smoking_status      1000 non-null   str    
 7   exercise_frequency  1000 non-null   str    
 8   bmi                 1000 non-null   float64
 9   blood_pressure      1000 non-null   float64
 10  diabetes            1000 non-null   bool

## Check Missing Values
Missing values can affect statistical calculations, so they are checked before analysis.

In [4]:
# Check missing values
print(df.isnull().sum())

record_id             0
age_group             0
age                   0
weight                0
gender                0
region                0
smoking_status        0
exercise_frequency    0
bmi                   0
blood_pressure        0
diabetes              0
hypertension          0
cholesterol_level     0
glucose_level         0
visit_date            0
dtype: int64


## Descriptive Statistics

Descriptive statistics provide information about the central tendency
and spread of numerical variables.

In [5]:
df.describe()

,age,weight,bmi,blood_pressure,cholesterol_level,glucose_level
count,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000
mean,50.380000,67.832000,24.585900,147.147500,223.732200,105.950200
std,18.378666,12.353234,5.430035,12.665471,26.325292,16.338605
min,18.000000,45.000000,13.500000,109.300000,135.800000,68.400000
25%,35.000000,59.000000,20.600000,138.100000,204.900000,94.200000
50%,50.000000,68.000000,24.200000,147.250000,222.200000,103.200000
75%,66.000000,76.000000,27.600000,156.100000,242.525000,114.400000
max,80.000000,108.000000,46.100000,184.000000,302.400000,156.800000


## Part A – Theoretical Foundation

### 1. What is Inferential Statistics?

Inferential statistics is a branch of statistics used to make conclusions
or predictions about a population based on sample data.

It includes methods such as hypothesis testing, confidence intervals,
and estimation.

### 2. What is Hypothesis Testing and its Components?

Hypothesis testing is a statistical method used to determine whether
there is enough evidence to support a claim about a population.

Its main components are Null Hypothesis (H₀), Alternative Hypothesis (H₁),
significance level, test statistic, p-value, critical value, and final decision.

### 3. Explain Confidence Interval and Critical Value

A confidence interval is a range of values within which a population
parameter is expected to lie with a specified confidence level.

A critical value is a boundary value used to determine whether the
test statistic falls in the rejection region.

### 4. Define p-value

A p-value represents the probability of obtaining a result as extreme
as the observed result, assuming that the null hypothesis is true.

If p-value is less than 0.05, the null hypothesis is generally rejected.

### 5. Differentiate Type I and Type II Errors

Type I Error occurs when a true null hypothesis is rejected.

Type II Error occurs when a false null hypothesis is not rejected.

### 6. Brief Description of Z-test, t-test, Chi-square Test and ANOVA

**Z-test:** Used mainly for testing means or proportions when the sample
size is large and population variance is known.

**t-test:** Used to compare the means of two groups, especially when
population standard deviation is unknown.

**Chi-square test:** Used to determine whether two categorical variables
are associated.

**ANOVA:** Used to determine whether there is a significant difference
among the means of three or more groups.

### 7. What is Covariance?

Covariance measures the direction in which two numerical variables
change together.

A positive covariance indicates that the variables tend to increase
together, while a negative covariance indicates an opposite movement.

### 8. What is Correlation?

Correlation measures the strength and direction of the relationship
between two numerical variables.

The Pearson correlation coefficient ranges from -1 to +1.
A value close to +1 indicates a strong positive relationship,
while a value close to -1 indicates a strong negative relationship.

# Part B – Data Analysis & Testing

## 1. Formulation of Hypotheses

Based on the health dataset, the following hypotheses are formulated.

### Hypothesis 1: Smoking Status and Diabetes

**H₀:** Smoking status has no significant association with diabetes.

**H₁:** Smoking status has a significant association with diabetes.

**Statistical Test:** Chi-square test

### Hypothesis 2: Glucose Level and Diabetes

**H₀:** The mean glucose level is the same for diabetic and
non-diabetic groups.

**H₁:** The mean glucose level is different between diabetic and
non-diabetic groups.

**Statistical Test:** Independent two-sample t-test

### Hypothesis 3: Age Group and Diabetes

**H₀:** The mean diabetes rate is the same across all age groups.

**H₁:** At least one age group has a significantly different diabetes rate.

**Statistical Test:** One-way ANOVA

#  2: Calculate 95% Confidence Interval

A 95% confidence interval is calculated for age, weight, BMI,
glucose level, and blood pressure.

The formula is:

CI = Mean ± Critical Value × Standard Error

In [6]:
# 95% Confidence Interval

alpha = 0.05

variables = [
    "age",
    "weight",
    "bmi",
    "glucose_level",
    "blood_pressure"
]

ci_results = []

for variable in variables:

    data = df[variable].dropna()

    n = len(data)
    mean = data.mean()
    std = data.std(ddof=1)

    standard_error = std / np.sqrt(n)

    critical_value = stats.t.ppf(
        1 - alpha / 2,
        n - 1
    )

    lower = mean - critical_value * standard_error
    upper = mean + critical_value * standard_error

    ci_results.append([
        variable,
        n,
        mean,
        std,
        standard_error,
        critical_value,
        lower,
        upper
    ])

ci_df = pd.DataFrame(
    ci_results,
    columns=[
        "Variable",
        "N",
        "Mean",
        "Standard Deviation",
        "Standard Error",
        "Critical Value",
        "CI Lower",
        "CI Upper"
    ]
)

ci_df

,Variable,N,Mean,Standard Deviation,Standard Error,Critical Value,CI Lower,CI Upper
0,age,1000,50.3800,18.378666,0.581184,1.962341,49.239518,51.520482
1,weight,1000,67.8320,12.353234,0.390644,1.962341,67.065424,68.598576
2,bmi,1000,24.5859,5.430035,0.171713,1.962341,24.248941,24.922859
3,glucose_level,1000,105.9502,16.338605,0.516672,1.962341,104.936313,106.964087
4,blood_pressure,1000,147.1475,12.665471,0.400517,1.962341,146.361548,147.933452


The 95% confidence intervals for the selected numerical variables were
successfully calculated.

The results are:

- **Age:** 95% CI = 49.2395 to 51.5205
- **Weight:** 95% CI = 67.0654 to 68.5986
- **BMI:** 95% CI = 24.2489 to 24.9229
- **Glucose Level:** 95% CI = 104.9363 to 106.9641
- **Blood Pressure:** 95% CI = 146.3615 to 147.9335

The critical t-value at the 95% confidence level is approximately
**1.9623**.

Therefore, these intervals represent the range of plausible population
means at the 95% confidence level.

## 3. Find the Critical Value and p-value

Find the critical value and p-value for the statistical tests and use them
to determine whether the null hypothesis should be rejected or not.

In [7]:
alpha = 0.05

# 1. Critical value for t-test
critical_t = stats.t.ppf(1 - alpha/2, df=len(df) - 2)

print("Critical t-value:", critical_t)

# 2. Critical value for Chi-square test
critical_chi2 = stats.chi2.ppf(1 - alpha, df=2)

print("Critical Chi-square value:", critical_chi2)

# 3. Critical value for ANOVA
critical_f = stats.f.ppf(1 - alpha, dfn=4, dfd=len(df) - 5)

print("Critical F-value:", critical_f)

Critical t-value: 1.962343846216334
Critical Chi-square value: 5.991464547107979
Critical F-value: 2.380875806928862


### Interpretation

At the 5% significance level:

- For the **t-test**, the p-value is less than 0.05. Therefore, we reject H₀.
- For the **Chi-square test**, the p-value is greater than 0.05. Therefore, we fail to reject H₀.
- For the **ANOVA test**, the p-value is less than 0.05. Therefore, we reject H₀.

Thus, the critical values and p-values help determine whether the
observed results are statistically significant.

## 4. Perform t-test

Compare the mean glucose levels between diabetic and non-diabetic
individuals using an independent two-sample t-test.

- **H₀:** There is no significant difference in mean glucose levels between diabetic and non-diabetic individuals.
- **H₁:** There is a significant difference in mean glucose levels between diabetic and non-diabetic individuals.

Formula:

t = (x̄₁ - x̄₂) / √(s₁²/n₁ + s₂²/n₂)

In [8]:
diabetic = df[df["diabetes"] == True]["glucose_level"].dropna()
non_diabetic = df[df["diabetes"] == False]["glucose_level"].dropna()

t_stat, p_value = stats.ttest_ind(
    diabetic,
    non_diabetic,
    equal_var=False
)

print("Diabetic Mean:", diabetic.mean())
print("Non-Diabetic Mean:", non_diabetic.mean())
print("t-statistic:", t_stat)
print("p-value:", p_value)

Diabetic Mean: 131.44820512820513
Non-Diabetic Mean: 99.77366459627329
t-statistic: 37.8708238049991
p-value: 3.2644385248126074e-115


### Interpretation

The independent two-sample t-test was performed to compare the mean
glucose levels between diabetic and non-diabetic individuals.

- **Diabetic Mean:** 131.4482
- **Non-Diabetic Mean:** 99.7737
- **t-statistic:** 37.8708
- **p-value:** 3.2644 × 10⁻¹¹⁵

Since the p-value is much less than 0.05, we reject the null hypothesis (H₀).

Therefore, there is a statistically significant difference in mean
glucose levels between diabetic and non-diabetic individuals.

## 5. Conduct a Chi-square Test

Conduct a Chi-square test to determine whether there is a significant
association between smoking status and diabetes.

- **H₀:** Smoking status has no significant association with diabetes.
- **H₁:** Smoking status has a significant association with diabetes.

In [9]:
# Create contingency table
contingency_table = pd.crosstab(
    df["smoking_status"],
    df["diabetes"]
)

print("Contingency Table:")
print(contingency_table)

# Perform Chi-square test
chi2_stat, p_value, degrees_of_freedom, expected = stats.chi2_contingency(
    contingency_table
)

# Critical value at 5% significance level
alpha = 0.05
critical_value = stats.chi2.ppf(
    1 - alpha,
    degrees_of_freedom
)

print("\nChi-square statistic:", chi2_stat)
print("Degrees of freedom:", degrees_of_freedom)
print("Critical value:", critical_value)
print("p-value:", p_value)

Contingency Table:
diabetes        False  True 
smoking_status              
Former Smoker     159     47
Non-Smoker        502    112
Smoker            144     36

Chi-square statistic: 2.091214509214106
Degrees of freedom: 2
Critical value: 5.991464547107979
p-value: 0.35147831773197674


### Interpretation

The Chi-square test was conducted to determine whether there is a
significant association between smoking status and diabetes.

- **Chi-square statistic:** 2.091
- **Degrees of freedom:** 2
- **Critical value:** 5.991
- **p-value:** 0.3515

Since the p-value is greater than 0.05, we fail to reject the null
hypothesis (H₀).

Therefore, there is no statistically significant association between
smoking status and diabetes in this dataset.

### 6. Perform an ANOVA Test

Perform a one-way ANOVA test to determine whether diabetes rates
differ significantly across different age groups.

- **H₀:** There is no significant difference in diabetes rates across age groups.
- **H₁:** There is a significant difference in diabetes rates across age groups.

Formula:

F = MS Between / MS Within

In [10]:
# Convert diabetes to numerical values
df["diabetes_numeric"] = df["diabetes"].astype(int)

# Create groups based on age_group
groups = [
    group["diabetes_numeric"].dropna()
    for name, group in df.groupby("age_group")
]

# Perform one-way ANOVA
f_stat, p_value = stats.f_oneway(*groups)

# Degrees of freedom
k = len(groups)
N = len(df)

df_between = k - 1
df_within = N - k

# Critical F-value
alpha = 0.05
critical_f = stats.f.ppf(
    1 - alpha,
    df_between,
    df_within
)

print("F-statistic:", f_stat)
print("p-value:", p_value)
print("Degrees of freedom (Between):", df_between)
print("Degrees of freedom (Within):", df_within)
print("Critical F-value:", critical_f)

F-statistic: 25.820554239424148
p-value: 2.1932158895740038e-20
Degrees of freedom (Between): 4
Degrees of freedom (Within): 995
Critical F-value: 2.380875806928862


### Interpretation

The one-way ANOVA test was performed to determine whether diabetes
rates differ significantly across different age groups.

- **F-statistic:** 25.821
- **p-value:** 2.19 × 10⁻²⁰
- **Degrees of freedom (Between):** 4
- **Degrees of freedom (Within):** 995
- **Critical F-value:** 2.381

Since the p-value is much less than 0.05, we reject the null hypothesis
(H₀).

Therefore, there is a statistically significant difference in diabetes
rates across the different age groups.

## 7. Calculate Covariance and Correlation

Calculate the covariance and Pearson correlation between age and BMI.

- **H₀:** There is no significant correlation between age and BMI.
- **H₁:** There is a significant correlation between age and BMI.

In [11]:
# Select Age and BMI
age_bmi = df[["age", "bmi"]].dropna()

# Covariance
covariance = age_bmi["age"].cov(age_bmi["bmi"])

# Pearson correlation
correlation, p_value = stats.pearsonr(
    age_bmi["age"],
    age_bmi["bmi"]
)

print("Covariance:", covariance)
print("Pearson Correlation:", correlation)
print("p-value:", p_value)

Covariance: 4.230588588588589
Pearson Correlation: 0.042392028534756765
p-value: 0.18041473871483202


### Interpretation

The covariance and Pearson correlation between age and BMI were
calculated.

- **Covariance:** 4.231
- **Pearson Correlation (r):** 0.0424
- **p-value:** 0.1804

The correlation coefficient is close to 0, indicating a very weak
positive relationship between age and BMI.

Since the p-value is greater than 0.05, we fail to reject the null
hypothesis (H₀).

Therefore, there is no statistically significant correlation between
age and BMI in this dataset.

### Result and Interpretation of Statistical Tests

The results of all statistical tests performed are summarized below.

| Test | Test Statistic | p-value | Decision | Interpretation |
|---|---:|---:|---|---|
| Independent t-test | 37.8708 | 3.2644 × 10⁻¹¹⁵ | **Reject H₀** | There is a significant difference in mean glucose levels between diabetic and non-diabetic individuals. |
| Chi-square test | 2.091 | 0.3515 | **Fail to Reject H₀** | There is no significant association between smoking status and diabetes. |
| One-way ANOVA | 25.821 | 2.19 × 10⁻²⁰ | **Reject H₀** | Diabetes rates differ significantly across age groups. |
| Pearson Correlation | 0.0424 | 0.1804 | **Fail to Reject H₀** | There is no statistically significant correlation between age and BMI. |

### Overall Interpretation

- **t-test:** Reject H₀ because p-value < 0.05.
- **Chi-square test:** Fail to reject H₀ because p-value > 0.05.
- **ANOVA:** Reject H₀ because p-value < 0.05.
- **Correlation:** Fail to reject H₀ because p-value > 0.05.

Therefore, the t-test and ANOVA showed statistically significant
results, while the Chi-square test and Pearson correlation did not show
statistically significant results.